### ISSUE TITLE
ebird: Checklist summary

### ISSUE THREAD

**Issue by 7-ate-9** [synrg/dronefly#56]
Similar to iNat observation but for eBird checklists
- Show observer name, date, time, location, checklist ID
- Show checklist comment, protocol, and duration
- Show number of species, number of photos and audio

**Comment by 7-ate-9** [comment]
It would be nice to show something about what species are on the list, but lists can have 50-100 species (although most will have 5-30) so I'm not sure what the best way to do that would be. Some possibilities:
- Show first 5 species on list
- Show first 3, ... show last 2
- Only show rare species

**Comment by dfloer** [comment]
Are you envisioning the command taking a checklist ID of the form "S59832077" (a checklist of mine from a couple months ago), or would there also need to be a way to discover the checklist (and IDs) too?

**Comment by synrg** [comment]
Yes, we'd need a way to discover those, but with the right API calls we can follow the linkage from the records returned by the `ebird hybrids` command.

If I recall correctly, I determined experimentally that the historic observations by date API call will give you the checklist IDs when you pass `detail=full` in the call:

https://documenter.getpostman.com/view/664302/S1ENwy59?version=latest#2d8c6ee8-c435-4e91-9f66-6d3eeb09edd2

So if you take the existing `ebird hybrids` command, for which both the regionCode and the date of recently observed observations for that region are known, that's all we need to do this call, e.g.

`https://api.ebird.org/v2/data/obs/US-MA/historic/2020/1/16?cat=hybrid&detail=full`

```
[
    {
        "checklistId": "CL23379",
        "comName": "Mallard x American Black Duck (hybrid)",
        "countryCode": "US",
        "countryName": "United States",
        "firstName": "chris",
        "hasComments": false,
        "hasRichMedia": false,
        "howMany": 1,
        "lastName": "roberts",
        "lat": 42.0332033,
        "lng": -72.5975322,
        "locId": "L330361",
        "locName": "Fannie Stebbins Wildlife Refuge",
        "locationPrivate": false,
        "obsDt": "2020-01-16 12:00",
        "obsId": "OBS852443863",
        "obsReviewed": false,
        "obsValid": true,
        "presenceNoted": false,
        "sciName": "Anas platyrhynchos x rubripes",
        "speciesCode": "x00004",
        "subId": "S63484808",
        "subnational1Code": "US-MA",
        "subnational1Name": "Massachusetts",
        "subnational2Code": "US-MA-013",
        "subnational2Name": "Hampden",
        "userDisplayName": "chris roberts"
    }
]
```

Just iterate over each unique date matched by the existing hybrids API call & that will give you all the `checklistId`'s you need to write this enhancement.

**Comment by synrg** [comment]
Yeah, it's this one: https://github.com/synrg/dronefly/issues/61

OK, so as to "similar to iNat observation", the way that typically works is just by sharing the URL of the checklist. Users already do that, so there's no "lookup" of the checklist ID needed. With the iNat command, we just Regex match the link & produce the observation summary from that.

**Comment by synrg** [comment]
> It would be nice to show something about what species are on the list, but lists can have 50-100 species (although most will have 5-30) so I'm not sure what the best way to do that would be.

Some sort of pagination seems reasonable. That could be provided by reaction buttons that update the emitted embed to show additional pages. If the list of species could be sorted so that "interesting" species are put first, then it makes it less likely any further interaction would be needed, giving people what they most likely want to see right away.
